### PR TITLE
Remove session singleton

### PR DESCRIPTION
**Pull request by vandenheuvel**
Solves #2286, requires https://github.com/Tribler/tribler/pull/3372.

### CODE REVIEW

**Comment by vandenheuvel** [comment]
retest this please

**Comment by vandenheuvel** [comment]
It appears the error on Windows 64-bit is unrelated to this PR: See #3367.

**Comment by devos50** [comment]
#3364 fixes the unit tests. I would like to merge that one first. After that's done, please rebase this PR and I will review it.

**Comment by devos50** [comment]
@vandenheuvel please rebase this PR on `devel` again. The tests should pass now.

File: Tribler/Core/Session.py [hunk 1]
```diff
@@ -60,32 +57,24 @@ def __init__(self, config=None, ignore_singleton=False, autoload_discovery=True)
...
-        self.ignore_singleton = ignore_singleton
         self.session_lock = NoDispersyRLock()
 
         self.config = config or TriblerConfig()
         self.get_ports_in_config()
         self.create_state_directory_structure()
 
-        if not config.get_megacache_enabled():
-            config.set_torrent_checking_enabled(False)
+        if not self.config.get_megacache_enabled():
```
Comment (vandenheuvel) [inline Tribler/Core/Session.py]: @devos50 I put these changes also in another PR, as this is a separate issue.

**Comment by vandenheuvel** [comment]
@devos50 Unit tests are working nicely. Ready!

**Comment by vandenheuvel** [comment]
retest this please

**Comment by vandenheuvel** [comment]
retest this please

**Comment by devos50** [comment]
Also, please ignore the Mac GUI tests, it's a known issue and requires physical access to the machine to reset it.

**Comment by vandenheuvel** [comment]
@devos50 Good point, will do.
